src/shared/conf-parser.c

```diff
@@ -497,29 +503,54 @@ static int config_parse_many_files(
                         return -ENOMEM;
         }
 
-        /* Get inodes for all drop-ins. Later we'll verify if main config is a symlink to one of them. If so,
-         * we skip reading main config file directly. */
         STRV_FOREACH(fn, files) {
-                _cleanup_free_ struct stat *st_dropin = NULL;
+                _cleanup_fclose_ FILE *f = NULL;
+                int fd;
 
-                st_dropin = new(struct stat, 1);
-                if (!st_dropin)
-                        return -ENOMEM;
+                f = fopen(*fn, "re");
+                if (!f) {
+                        if (errno == ENOENT)
+                                continue;
 
-                r = RET_NERRNO(stat(*fn, st_dropin));
-                if (r < 0 && r != -ENOENT)
-                        return r;
+                        return -errno;
+                }
 
-                r = set_ensure_consume(&inodes, &inode_hash_ops, TAKE_PTR(st_dropin));
-                if (r < 0)
+                fd = fileno(f);
```

Comment:
I'd like to move this line just before `if (fd >= 0)`.

Reply:
I've dropped the condition on fd >= 0. I placed it here because otherwise I need to separate `hashmap_ensure_put` from `TAKE_PTR`, which IMO makes the code harder to track.